doc/architecture/backend.md

```diff
@@ -0,0 +1,68 @@
+# Backend Architecture
+
+## Code documentation
+
+The code documentation of the backend can be built using `cargo doc --open`.
+
+You can find a more detailed explanation of which modules do what there.
+
+## API documentation
+
+The API documentation can be viewed by running the backend with `cargo run` and then navigating to <http://localhost:8080/doc/api/swagger/ui/>.  
+It is automatically built using [utoipa](https://github.com/juhaku/utoipa).
+
+## Structure
+
+The backend is split using a 3-layer architecture with controller, service and persistence layer.
+
+```mermaid
+graph LR;
+    Controller<-->Service;
+    Service<-->Persistence;
```

Comment:
Should be named like our modules are named (Persistence -> Db?). Do we really don't have a single point where a Controller interacts with Persistence? What about our "config" and "model"?

Reply:
I don't agree with that. The architecture itself is different from how it is implemented/how the code looks. The individual sections then describe where the code can be found, but is the layer should still be called Persistence. This is more about naming convention so that developers easily understand the architecture.

The controller layer should never actually interact with the persistence layer and I don't think they do in our code. The model folder simply represents our data, but has not much to do with the actual 3-layer architecture. We use DTOs and entities and that way they are separated. 
Config is the configuration of the server. It has nothing to do with the 3 layers (although there might be code in there that touch all layer, but again that is fine as the controller and persistence layer themselves don't interact).

Reply:
Does it even make sense then to have a diagram? 
The modules themselves don't really interact with each other, but the code that runs inside them does. At least I have no idea how I would connect `config` or `db` to `service` or `controller`. Then there is the difference between the module `db` and persistence as I defined it. The module `db` simply contains a bit of utility functions for the database, but the actual calls to the database are made from the `model/entity/` folder.
So the question is how is module defined in arc42? Is it the structure of the code in the filesystem (in which case Rust doc handles it) or the structure of the running system (then I don't think I should describe the module `db` as it isn't the one interacting with the system)?

If I should describe the structure/modules wouldn't it be better to do something like the following:
```
config: Contains the servers configuration
controller: Contains the API endpoints
service: Contains the service layer
...
```
but then that is exactly the same as the code doc Rust generates.

Reply:
Maybe the intent of arc would be more something like:

```mermaid
graph LR;
    Plants<-->Plantings;
    Seed;
```

That way the actual components are described. But then our system isn't complicated enough for something like that to make much sense. I don't even think any of those actually interact with each other.

Reply:
> Does it even make sense then to have a diagram?

Yes, to illustrate dependencies a diagram is perfect.

> The modules themselves don't really interact with each other,

Of course they do. The black box view shows if we use another module from any module, i.e., how the depend on each other. If an actual interaction takes place (calling functions) or only macros etc. are used does not really matter.

And we need to be honest with ourself, we need to either document or fix the deps we have.

> The module db simply 

Please fix the PR to contain such information.

> So the question is how is module defined in arc42?

See above: like in every architectural description (it is not specific to arc42): It needs to show dependencies between modules.

> structure of the running system (then I don't think I should describe the module db as it isn't the one interacting with the system)?

For the running system (the order of interactions) there is the runtime diagram: https://docs.arc42.org/section-6/

But this doesn't make much sense to be drawn for backend only, it should show interactions between frontend and backend.

I hope @Bushuo can jump in here. I crated #326 for the next step.

> but then that is exactly the same as the code doc Rust generates.

Obviously, architecture is only a high-level description of the code.